### PR TITLE
AF-593: Decouple kie-wb-common-datamodel from Drools

### PR DESCRIPTION
**Pull request by mbarkley**

### CODE REVIEW

**Comment by mbarkley** [comment]
Related:
https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/541
https://github.com/kiegroup/droolsjbpm-knowledge/pull/263
https://github.com/kiegroup/drools/pull/1444
https://github.com/kiegroup/jbpm/pull/973
https://github.com/kiegroup/droolsjbpm-integration/pull/1163
https://github.com/kiegroup/kie-wb-common/pull/1133
https://github.com/kiegroup/drools-wb/pull/596
https://github.com/kiegroup/jbpm-designer/pull/654
https://github.com/kiegroup/jbpm-wb/pull/864
https://github.com/kiegroup/optaplanner-wb/pull/218

**Comment by ederign** [comment]
jenkins please execute full downstream build

File: uberfire-project-datamodel/uberfire-project-datamodel-api/src/main/java/org/appformer/project/datamodel/oracle/ExtensionKind.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (C) 2017 Red Hat, Inc. and/or its affiliates.
```
Comment (manstis) [inline uberfire-project-datamodel/uberfire-project-datamodel-api/src/main/java/org/appformer/project/datamodel/oracle/ExtensionKind.java]: Surplus ```(C)```

File: uberfire-project-datamodel/uberfire-project-datamodel-api/src/main/java/org/appformer/project/datamodel/oracle/PackageDataModelOracle.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2012 Red Hat, Inc. and/or its affiliates.
```
Comment (manstis) [inline uberfire-project-datamodel/uberfire-project-datamodel-api/src/main/java/org/appformer/project/datamodel/oracle/PackageDataModelOracle.java]: 2017

File: uberfire-project-datamodel/uberfire-project-datamodel-commons/src/main/java/org/appformer/project/datamodel/commons/oracle/PackageDataModelOracleImpl.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2015 Red Hat, Inc. and/or its affiliates.
```
Comment (manstis) [inline uberfire-project-datamodel/uberfire-project-datamodel-commons/src/main/java/org/appformer/project/datamodel/commons/oracle/PackageDataModelOracleImpl.java]: 2017

File: uberfire-project-datamodel/uberfire-project-datamodel-commons/src/main/java/org/appformer/project/datamodel/commons/types/ClassTypeResolver.java [hunk 4]
```diff
@@ -0,0 +1,383 @@
+/*
+ * Copyright 2005 Red Hat, Inc. and/or its affiliates.
```
Comment (manstis) [inline uberfire-project-datamodel/uberfire-project-datamodel-commons/src/main/java/org/appformer/project/datamodel/commons/types/ClassTypeResolver.java]: 2017

File: uberfire-project-datamodel/uberfire-project-datamodel-commons/src/main/java/org/appformer/project/datamodel/commons/types/TypeResolver.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2010 Red Hat, Inc. and/or its affiliates.
```
Comment (manstis) [inline uberfire-project-datamodel/uberfire-project-datamodel-commons/src/main/java/org/appformer/project/datamodel/commons/types/TypeResolver.java]: 2017

File: uberfire-project-datamodel/uberfire-project-datamodel-commons/src/main/java/org/appformer/project/datamodel/commons/util/MVELEvaluator.java [hunk 6]
```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) 2017 Red Hat, Inc. and/or its affiliates.
```
Comment (manstis) [inline uberfire-project-datamodel/uberfire-project-datamodel-commons/src/main/java/org/appformer/project/datamodel/commons/util/MVELEvaluator.java]: Surplus ```(C)```

File: uberfire-project-datamodel/uberfire-project-datamodel-commons/src/main/java/org/appformer/project/datamodel/commons/util/RawMVELEvaluator.java [hunk 7]
```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (C) 2017 Red Hat, Inc. and/or its affiliates.
```
Comment (manstis) [inline uberfire-project-datamodel/uberfire-project-datamodel-commons/src/main/java/org/appformer/project/datamodel/commons/util/RawMVELEvaluator.java]: Surplus ```(C)```

**Comment by mbarkley** [comment]
Jenkins, please execute full downstream build.

**Comment by mbarkley** [comment]
@manstis Thanks for catching those. Working on them now.

**Comment by mbarkley** [comment]
@manstis Issues fixed.

@ederign There are test failures in some of the downstream pull requests. I will retry the full build after those are fixed.

**Comment by mbarkley** [comment]
Jenkins, please execute full downstream build.

**Comment by mbarkley** [comment]
@manstis @ederign Please do not merge this yet. There is a missing PR I need to send to kie-wb-distributions.

**Comment by mbarkley** [comment]
Jenkins, please execute full downstream build.

**Comment by mbarkley** [comment]
Jenkins, please execute full downstream build.

**Comment by mbarkley** [comment]
@ederign There appear to be GitHub API issues so the downstream build is not working.

AFAICT, the only repo I have not built fully (with tests) that does not have a PR sent to it is optaplanner, so I'm building that locally now to make sure everything is okay there. If that works, can we approve this?

**Comment by mbarkley** [comment]
(Optaplanner builds locally with the changes.)

File: uberfire-project-datamodel/uberfire-project-datamodel-commons/pom.xml [hunk 8]
```diff
@@ -70,7 +74,7 @@
         <configuration>
           <instructions>
             <Export-Package>
-              org.appformer.project.datamodel.commons.*
+              org.appformer.project.datamodel.commons*
```
Comment (ederign) [inline uberfire-project-datamodel/uberfire-project-datamodel-commons/pom.xml]: @mbarkley this is intentional right?

File: uberfire-project-datamodel/uberfire-project-datamodel-commons/src/main/java/org/appformer/project/datamodel/commons/types/ClassTypeResolver.java [hunk 9]
```diff
@@ -0,0 +1,383 @@
...
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+
+public class ClassTypeResolver
+    implements
+    TypeResolver {
+    private String                           defaultPackagName;
```
Comment (ederign) [inline uberfire-project-datamodel/uberfire-project-datamodel-commons/src/main/java/org/appformer/project/datamodel/commons/types/ClassTypeResolver.java]: Please check the code style.

File: uberfire-project-datamodel/uberfire-project-datamodel-commons/pom.xml (hunk 8 above)
Comment (mbarkley) [reply uberfire-project-datamodel/uberfire-project-datamodel-commons/pom.xml]: Yeah, this was a change I made when I was trying to get the kie-karaf-itests to pass. It turns out to not make a difference, but it matches the way some of the other patterns are specified (without the dot near the end).

I can get rid of it if you want, as I don't think it makes any difference.

File: uberfire-project-datamodel/uberfire-project-datamodel-commons/src/main/java/org/appformer/project/datamodel/commons/types/ClassTypeResolver.java (hunk 9 above)
Comment (mbarkley) [reply uberfire-project-datamodel/uberfire-project-datamodel-commons/src/main/java/org/appformer/project/datamodel/commons/types/ClassTypeResolver.java]: I am still using Eclipse. AFAIK there is still no way to do proper formatting with the new code-styles in Eclipse, and I don't know where there is any document other than an IntelliJ config that describes the code-style, so I'm not sure how to fix this.

**Comment by mbarkley** [comment]
@ederign Turns out you can run the IntelliJ formatter headless, so I fixed the formatting in that file.

**Comment by mbarkley** [comment]
@ederign Fixed the formatting of the `ClassTypeResolver` and removed the unnecessary change to the `Export-Package` of uberfire-project-datamodel-commons. Everything good now?

**Comment by mbarkley** [comment]
Jenkins, please retest this.

**Pull request merged**